app/controllers/Application.java

```diff
@@ -1043,4 +1134,244 @@ public static Result restJobGraphData(String jobDefId) {
 
     return ok(new Gson().toJson(datasets));
   }
+
+
+  /**
+   * The data for plotting the job history graph using time and resource metrics. While plotting the job history
+   * graph an ajax call is made to this to fetch the graph data.
+   *
+   * Data Returned:
+   * <pre>
+   * [
+   *  {
+   *    "flowtime": 1461234105456,
+   *    "runtime": 2312107,
+   *    "waittime": 118879,
+   *    "resourceused": 304934912,
+   *    "resourcewasted": 172913,
+   *    "jobmetrics": [
+   *      {
+   *        "stageid": "application_1458194917883_1587177",
+   *        "runtime": 642986,
+   *        "waittime": 14016,
+   *        "resourceused": 277352448,
+   *        "resourcewasted": 0
+   *    }],
+   *  },
+   *  {
+   *    "flowtime": 1461237538639,
+   *    "runtime": 2155354,
+   *    "waittime": 112187,
+   *    "resourceused": 293096448,
+   *    "resourcewasted": 400461,
+   *    "jobmetrics": [
+   *      {
+   *        "stageid": "application_1458194917883_1589302",
+   *        "runtime": 548924,
+   *        "waittime": 16903,
+   *        "resourceused": 266217472,
+   *        "resourcewasted": 0
+   *      }]
+   *  }
+   *  ]
+   *
+   * </pre>
+   */
+  public static Result restJobMetricsGraphData(String jobDefId) {
+    JsonArray datasets = new JsonArray();
+    if (jobDefId == null || jobDefId.isEmpty()) {
+      return ok(new Gson().toJson(datasets));
+    }
+
+    // Fetch available flow executions with latest JOB_HISTORY_LIMIT mr jobs.
+    List<AppResult> results = AppResult.find
```

Comment:
this query seems to be repeated in multiple places in the code. Can you encapsulate it in a single function and call from everywhere ?